.github/workflows/test.yml

```diff
@@ -104,53 +104,44 @@ jobs:
           cat pkgs.txt.part.${{ matrix.part }} | xargs go test -mod=readonly -timeout 30m -coverprofile=${{ matrix.part }}profile.out -covermode=atomic -tags='ledger test_ledger_mock'
         if: env.GIT_DIFF
       - uses: actions/upload-artifact@v4
+        if: env.GIT_DIFF
         with:
           name: '${{ github.sha }}-${{ matrix.part }}-coverage'
           path: ./${{ matrix.part }}profile.out
 
-  upload-coverage-report:
+  repo-analysis:
     runs-on: ubuntu-latest
-    needs: tests
+    needs: [tests]
     steps:
       - uses: actions/checkout@v4
       - uses: technote-space/get-diff-action@v6.1.2
+        id: git_diff
         with:
           PATTERNS: |
-            **/**.go
+            **/*.go
             go.mod
             go.sum
+            **/go.mod
+            **/go.sum
       - uses: actions/download-artifact@v4
+        if: env.GIT_DIFF
         with:
           name: '${{ github.sha }}-00-coverage'
-        if: env.GIT_DIFF
       - uses: actions/download-artifact@v4
+        if: env.GIT_DIFF
         with:
           name: '${{ github.sha }}-01-coverage'
-        if: env.GIT_DIFF
       - uses: actions/download-artifact@v4
+        if: env.GIT_DIFF
         with:
           name: '${{ github.sha }}-02-coverage'
-        if: env.GIT_DIFF
       - uses: actions/download-artifact@v4
-        with:
-          name: '${{ github.sha }}-03-coverage'
-        if: env.GIT_DIFF
-      - run: |
-          cat ./*profile.out | grep -v "mode: atomic" >> coverage.txt
         if: env.GIT_DIFF
-      - name: filter out DONTCOVER
-        run: |
-          excludelist="$(find ./ -type f -name '*.go' | xargs grep -l 'DONTCOVER')"
-          excludelist+=" $(find ./ -type f -name '*.pb.go')"
-          excludelist+=" $(find ./ -type f -name '*.pb.gw.go')"
-          excludelist+=" $(find ./ -type f -path './tests/mocks/*.go')"
-          for filename in ${excludelist}; do
-            filename=$(echo $filename | sed 's/^./github.com\/cosmos\/cosmos-sdk/g')
-            echo "Excluding ${filename} from coverage report..."
-            sed -i.bak "/$(echo $filename | sed 's/\//\\\//g')/d" coverage.txt
-          done
-        if: env.GIT_DIFF
-      - uses: codecov/codecov-action@v4
         with:
-          file: ./coverage.txt
-        if: env.GIT_DIFF
+          name: '${{ github.sha }}-03-coverage'
+      - name: sonarcloud
+        if: ${{ env.GIT_DIFF && !github.event.pull_request.draft }}
```

Comment:
are we missing the null check for `secrets.SONAR_TOKEN` here?

Reply:
thinking a bit more, we probably don't need that check at all as we're passing it in in the same workflow, we would just be checking that `secrets.SONAR_TOKEN` is set basically, maybe unnecessary.